pkg/cloudprovider/providers/rackspace/rackspace.go

```diff
@@ -391,3 +467,145 @@ func (os *Rackspace) GetZone() (cloudprovider.Zone, error) {
 
 	return cloudprovider.Zone{Region: os.region}, nil
 }
+
+// Create a volume of given size (in GiB)
+func (rs *Rackspace) CreateVolume(name string, size int, tags *map[string]string) (volumeName string, err error) {
+	return "", errors.New("unimplemented")
+}
+
+func (rs *Rackspace) DeleteVolume(volumeName string) error {
+	return errors.New("unimplemented")
+}
+
+// Attaches given cinder volume to the compute running kubelet
+func (rs *Rackspace) AttachDisk(diskName string) (string, error) {
+	instanceID, err := readInstanceID()
+	if err != nil {
+		return "", err
+	}
+
+	disk, err := rs.getVolume(diskName)
+	if err != nil {
+		return "", err
+	}
+
+	compute, err := rs.getComputeClient()
+	if err != nil {
+		return "", err
+	}
+
+	if len(disk.Attachments) > 0 {
```

Comment:
I'd just remove this client-side check and let the later `volumeattach.Create` fail if attaching is inappropriate.  It's probably just fine on rackspace, but some of the wider openstack cinder drivers support "multiattach".